pkgs/os-specific/linux/iproute/default.nix

```diff
@@ -26,8 +26,9 @@ stdenv.mkDerivation rec {
     "HDRDIR=$(TMPDIR)/include/iproute2" # Don't install headers
   ];
 
+  # enable iproute2 module if you want this folder to be created
   buildFlags = [
-    "CONFDIR=/etc/iproute2"
+    "CONFDIR=/run/iproute2"
```

Comment:
You can share this with the module by using something like ```config.iproute2.confdir or "/etc/iproute2"```

Reply:
I was looking for a way to unify the setting too but not sure how to do it. I just tried sthg but it fails. Any suggestion ? Thanks for your patience

Reply:
Yes I think you would do this like:

```
"CONFDIR=${config.iproute2.confdir or "/run/iproute2"}"
```

& then add config as an arg.

Also add to your module implementation:

```
nixpkgs.config.iproute2.confdir = "/run/iproute2"
```

Reply:
just making sure:
- does it have to be a toplevel config.iproute2 or is the current config.networking.iproute2 ok ? 
- I already have confDir as an option ? isn't that enough ?